lib/mqtt.c

```diff
@@ -737,6 +749,35 @@ static CURLcode mqtt_done(struct Curl_easy *data,
   return CURLE_OK;
 }
 
+/* we ping every 30 seconds to avoid being disconnected by the server */
+static CURLcode mqtt_ping(struct Curl_easy *data)
+{
+  CURLcode result = CURLE_OK;
+  struct connectdata *conn = data->conn;
+  struct mqtt_conn *mqtt = &conn->proto.mqtt;
+  struct MQTT *mq = data->req.p.mqtt;
+
+  if(mqtt->state == MQTT_FIRST &&
+     mq->pingsent == 0 &&
+     data->set.tcp_keepalive) {
```

Comment:
This is not TCP keepalive and I don't think you should include that in the check. It is separate and independent. We can possibly consider an option in the future to control the use of this ping.

Reply:
I used keepalive flag to avoid making a new flag just for this.
We could have an extra option.